bin/genesis

```diff
@@ -3655,6 +3691,296 @@ sub {
 	explain "New environment $name provisioned.\n";
 });
 
+
+
+# }}}
+# genesis ci - generate pipeline YAML template file to be used by genesis repipe to generate a pipeline file. {{{
+
+command("ci", <<EOF,
+genesis v$VERSION
+USAGE: genesis ci [--name ci.yml]
+
+OPTIONS
+$GLOBAL_USAGE
+      --name        The name of the yaml file to generate, default is `ci.yml'
+
+EOF
+sub {
+	my %options = (
+		 name => "ci.yml",
+	);
+
+	options(\@_, \%options, qw/
+		name=s
+	/);
+	usage(1) if @_ != 0;
+
+	my $name = $options{name} || "ci.yml";
+
+	at_exit(sub {
+		unlink "$name" unless $? == 0;
+	});
+
+	check_prereqs;
+	explain "Generating new CI YAML Template #C{$name}...\n";
+
+	my %pipeline;
+
+	$pipeline{name} = prompt_for_line("What is the name of your pipeline? \n It will display in Concourse, ex: cf-deployments.");
+
+	$pipeline{public} = (prompt_for_boolean("Would you like to make the pipeline visible to everyone? \nSetting this to yes will make the Concourse pipeline (and all logs related to deployments) public. This is off by default.", "no") ? "yes" : "no");
+
+	$pipeline{tagged} = (prompt_for_boolean("Would you like to use tags on each job? If you do not know why you need it, then you do not need it. Default is no.","no") ? "yes" : "no");
+
+	$pipeline{debug} = (prompt_for_boolean("Would you like to use debug on bosh jobs? Default: no." ,"no") ? "yes" : "no");
+
+	$pipeline{skip_upkeep} = (prompt_for_boolean("Would you like to automatically update stemcells? If set, disables the automatic stemcell updating behavior of the pipeline. Default is no.","no") ? "yes" : "no");
+
+	$pipeline{git} = prompt_for_git();
+
+	$pipeline{notifications} = prompt_for_notifications();
+
+	$pipeline{locker} = prompt_for_locker();
+
+	$pipeline{vault} = prompt_for_vault();
+
+	$pipeline{stemcells} = prompt_for_block("What is(are) the stemcell(s) you need for all your deployments in this pipeline? \nOne stemcell per line, for example you can input the bosh-lite stemcell like the following example and then hit enter.\nbosh-lite: bosh-warden-boshlite-ubuntu-trusty-go_agent\nNote:\"bosh-lite\" is the short name you want to call this stemcell which will be used to configure stemcell for your \"boshes\" later; \"bosh-warden-boshlite-ubuntu-trusty-go_agent\" is the stemcell name from 'bosh.io'.\n");
```

Comment:
Is there a better way to format this code, perhaps via a heredoc?  Long lines are very hard to read.

Reply:
I will do some research on it , thanks for your feedback !!